internal/serialization/portable_serializer_test.go

```diff
@@ -528,5 +529,131 @@ func TestPortableSerializer_NestedPortableVersion(t *testing.T) {
 	if !reflect.DeepEqual(deserializedParent, p) {
 		t.Error("nested portable version is wrong")
 	}
+}
+
+const (
+	factoryID1 = 1
+	factoryID2 = 2
+)
+
+type MyPortable1 struct {
+	stringField string
+}
+
+func (MyPortable1) FactoryID() (factoryID int32) {
+	return factoryID1
+}
+
+func (MyPortable1) ClassID() (classID int32) {
+	return 1
+}
+
+func (p MyPortable1) WritePortable(writer serialization.PortableWriter) {
+	writer.WriteString("stringField", p.stringField)
+}
+
+func (p *MyPortable1) ReadPortable(reader serialization.PortableReader) {
+	p.stringField = reader.ReadString("stringField")
+}
+
+type MyPortable2 struct {
+	intField int32
+}
+
+func (MyPortable2) FactoryID() (factoryID int32) {
+	return factoryID2
+}
+
+func (MyPortable2) ClassID() (classID int32) {
+	return 1
+}
+
+func (p MyPortable2) WritePortable(writer serialization.PortableWriter) {
+	writer.WriteInt32("intField", p.intField)
+}
+
+func (p *MyPortable2) ReadPortable(reader serialization.PortableReader) {
+	p.intField = reader.ReadInt32("intField")
+}
+
+type MyPortableFactory1 struct{}
+
+func (MyPortableFactory1) Create(classID int32) (instance serialization.Portable) {
+	if classID == 1 {
+		return &MyPortable1{}
+	}
+	return
+}
+
+func (MyPortableFactory1) FactoryID() int32 {
+	return factoryID1
+}
+
+type MyPortableFactory2 struct{}
+
+func (MyPortableFactory2) Create(classID int32) (instance serialization.Portable) {
+	if classID == 1 {
+		return &MyPortable2{}
+	}
+	return
+}
+
+func (*MyPortableFactory2) FactoryID() int32 {
+	return factoryID2
+}
 
+func TestClassesWithSameClassIdInDifferentFactories(t *testing.T) {
```

Comment:
Could you also add a comment about the corresponding Java test? Something like:
```
// ported from: com.hazelcast.internal.serialization.impl.portable.ExplicitClassDefinitionRegistrationTest#test_classesWithSameClassIdInDifferentFactories
```
(I've used copy reference on IntellJ)

Reply:
Added, [cd443da](https://github.com/hazelcast/hazelcast-go-client/pull/758/commits/cd443da556767f116d96dd7a518151744bad043e).

Reply:
Could you add it below the function signature and keep it in a single line? Like:
```
func TestClassesWithSameClassIdInDifferentFactories(t *testing.T) {
    // ported from: com.hazelcast.internal.serialization.impl.portable.ExplicitClassDefinitionRegistrationTest#test_classesWithSameClassIdInDifferentFactories
```

Reply:
Fixed, [5dca40b](https://github.com/hazelcast/hazelcast-go-client/pull/758/commits/5dca40bd3b2c69bfe4310bb3c293c71d358b2116)